go-controller/pkg/ovn/master.go

```diff
@@ -602,12 +603,53 @@ func (oc *DefaultNetworkController) syncNodes(kNodes []interface{}) error {
 	if err != nil {
 		return fmt.Errorf("failed to get node logical switches which have other-config set: %v", err)
 	}
+
+	staleNodes := sets.NewString()
 	for _, nodeSwitch := range nodeSwitches {
 		if !foundNodes.Has(nodeSwitch.Name) {
-			if err := oc.cleanupNodeResources(nodeSwitch.Name); err != nil {
-				return fmt.Errorf("failed to cleanup node resources:%s, err:%v", nodeSwitch.Name, err)
+			staleNodes.Insert(nodeSwitch.Name)
+		}
+	}
+
+	// Find stale external logical swiches, based on well known prefix and node name
+	lookupExtSwFunction := func(item *nbdb.LogicalSwitch) bool {
+		nameSplit := strings.Split(item.Name, types.ExternalSwitchPrefix)
```

Comment:
maybe it's simpler to do
```
if strings.HasPrefix(item.Name, types.ExternalSwitchPrefix) {
    nodeName := strings.TrimPrefix(item.Name, types.ExternalSwitchPrefix)
}
```

Reply:
I like it, mostly! ;) TrimPrefix calls HasPrefix, so it feels a little redundant. 
Are you okay if we did this:
```
lookupExtSwFunction := func(item *nbdb.LogicalSwitch) bool {
		nodeName := strings.TrimPrefix(item.Name, types.ExternalSwitchPrefix)
		if nodeName != item.Name && len(nodeName) > 0 && !foundNodes.Has(nodeName) {
			staleNodes.Insert(nodeName)
			return true
		}
		return false
	}
```  ?